packages/themes/build.js

```diff
@@ -0,0 +1,243 @@
+        semantic: 'theme',
+        'light-mode': 'light',
+        'dark-mode': 'dark'
+      });
+
+      return parsedJsonObj;
+    } catch (error) {
+      console.error(error);
+    }
+  }
+});
+
+console.log('Build started...');
+console.log('\n==============================================');
+
+/*
+ THESE ARE CONTITIONS TO FILTER ON AND A HELPER FUNCTION TO CREATE MATCHING FILTERS
+ These are a convenience to avoid long filter strings such as
+ token.attributes.category === 'global' || token.attributes.category === 'spacing' || token....
+ The objects defined here are allowed through - everything else is filtered out. Acts as a yes gate
+ The not: array allows for partial filtering of object keys. Acts as a no gate
+*/
+const conditionsCssLightTheme = [{ category: 'theme', type: 'light' }];
+
+const conditionsCssDarkTheme = [{ category: 'theme', type: 'dark' }];
+
+const conditionsCssGlobal = [
+  // token.attributes.category === 'global' || token.attributes.category === 'spacing'
+  { category: 'global', subitem: { not: ['seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsTw = [
+  { category: 'theme', type: 'light' },
+  { category: 'global', subitem: { not: ['light', 'dark', 'seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsJs = [
+  { category: 'theme' },
+  {
+    category: 'global',
+    type: { not: ['spacing'] },
+    item: { not: ['mode'] },
+    subitem: { not: ['seed'] }
+  }
+];
+
+const matchesCondition = (token, condition) => {
+  return Object.entries(condition).every(([key, value]) => {
+    if (typeof value === 'object' && value.not) {
+      return !value.not.includes(token.attributes[key]);
+    }
+    return token.attributes[key] === value;
+  });
+};
+
+const tokenMatchesAnyCondition = (token, conditions) => {
+  return conditions.some((condition) => matchesCondition(token, condition));
+};
+
+const createMatcher = (conditions) => (token) => tokenMatchesAnyCondition(token, conditions);
+
+// REGISTER THE CUSTOM FILTERS USING OUR MATCHING CONDITIONS
+
+StyleDictionary.registerFilter({
+  name: 'cssGlobalFilter',
+  matcher: (token) => createMatcher(conditionsCssGlobal)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssLightThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssLightTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssDarkThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssDarkTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'twFilter',
+  matcher: (token) => createMatcher(conditionsTw)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'jsFilter',
+  matcher: (token) => createMatcher(conditionsJs)(token)
+});
+
+/**
+ * Custom format that handle reference in css variables
+ */
+const transformString = (str, replace = 'theme', regex = /.*-color/) => {
+  // Use a regular expression to replace everything before regex with replace
+  return str.replace(regex, replace);
+};
+
```

Comment:
This is misleading: the function replaces the regex match, **not** everything before the regex match.
(of course, if the regex begins with `.*`, then it that part of the regex will match any amount of anything before the rest of the match, but whatever it matches is part of the regex match).

All this function does is supply default values to `str.replace`: I'd remove it and use `str.replace()` directly. This has the advantage of being more explicit about what substitutions are being done.